hal_st/stm32fxxx/FlashInternalStm.cpp

```diff
@@ -105,8 +140,9 @@ namespace hal
         }
     }
 
-#ifdef STM32WBA
+#if defined(STM32WBA) || defined(STM32H5)
```

Comment:
unrelated to your change, but the variable `dataSize` on line 152 is never used.

I think the intention was to read once, and use that variable on line 155. But 'chunk' is updated every loop, so it has be read explicitly every loop.

Also, I think we can consolidate this non-template member function and the member function template in a third one, accepting a std::size_t as the first argument for alignment.

Thus we would get:

```
        template<std::size_t alignment, uint32_t flashType>
        void AlignedWriteBuffer(infra::ConstByteRange buffer, uint32_t address);

        template<typename alignment, uint32_t flashType>
        void AlignedWriteBuffer(infra::ConstByteRange buffer, uint32_t address);

#if defined(STM32WBA) || defined(STM32H5)
        void AlignedWriteBuffer(infra::ConstByteRange buffer, uint32_t address);
#endif
```

Where `AlignedWriteBuffer<typename alignment>`  simply calls `AlignedWriteBuffer<std::size_t>` with `sizeof(alignment)`;

This allows the non-templated version to also re-use that same implementation.

@richardapeters any thoughts on this?

Reply:
I would indeed be in favor of consolidating the current functions into one function if possible; we could hopefully then also get rid of the member function template; this is an odd place to have member function templates, and the only purpose seems to be to avoid some duplication.

Reply:
After digging into it, it's not really possible to consolidate those functions into one because the semantics of `HAL_FLASH_Program` changed with the different MCU types:
- `stm32f4xx`, `stm32f7xx`, `stm32g0xx`, `stm32g4xx`, `stm32wbxx` expects the last parameter to be the actual data to be programmed
- `stm32wbaxx`,  `stm32h5xx` expects the last parameter to be a pointer to data to be programmed

Reply:
ugh, stupid st-HAL

Reply:
So we keep it as it is?

Reply:
I guess so :-/ I still think we can consolidate them, but then there would be a ifdef within the  while() loop. Food for thought for another time I guess.

I mean, the beginning of the function and the beginning of the while loop are the same.

Reply:
I know, in the end its a trade-off between number of methods vs number of ifdefs. It should be kept readable in the end.